android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/SyncPerformanceIntTestsHarness.kt

```diff
@@ -0,0 +1,289 @@
+            mapOf(
+                "runId" to runId,
+                "name" to BsonString(this.testName),
+                "dataProbeGranularityMs" to BsonInt64(this.dataProbeGranularityMs),
+                "numOutliersEachSide" to BsonInt32(this.numOutliersEachSide),
+                "numIters" to BsonInt32(this.numIters),
+                "date" to BsonDateTime(Date().time),
+                "sdk" to BsonString("android"),
+                "results" to BsonArray()
+            )
+        )
+    }
+}
+
+data class MemoryAndThreadCount(val memoryData: List<Long>, val threadData: List<Int>)
+
+class SyncPerformanceIntTestsHarness : BaseStitchAndroidIntTest() {
+    // Private constants
+    private val mongodbUriProp = "test.stitch.mongodbURI"
+    private val stitchAPIKeyProp = "test.stitch.androidPerfStitchAPIKey"
+
+    private val stitchOutputAppName = "android-sdk-perf-testing-yuvef"
+    private val stitchOutputDbName = "performance"
+    private val stitchOutputCollName = "results"
+
+    internal val stitchTestDbName = "performance"
+    internal val stitchTestCollName = "rawTestColl"
+    internal var stitchTestHost = ""
+
+    private val transport by lazy { OkHttpInstrumentedTransport() }
+    private val cpuMonitor by lazy { CPUMonitor() }
+
+    // Private variables
+    internal lateinit var outputClient: StitchAppClient
+    internal lateinit var outputMongoClient: RemoteMongoClient
+    internal lateinit var outputColl: RemoteMongoCollection<Document>
+    internal lateinit var mdbService: Apps.App.Services.Service
+    internal lateinit var mdbRule: RuleResponse
+
+    // Public variables
+    lateinit var testClient: StitchAppClient
+    var testDbName = ObjectId().toHexString()
+    var testCollName = ObjectId().toHexString()
+    var testNamespace = MongoNamespace(testDbName, testCollName)
+    lateinit var testMongoClient: RemoteMongoClient
+    lateinit var testColl: RemoteMongoCollection<Document>
+    lateinit var testUserId: String
+
+    val testDataSynchronizer: DataSynchronizer
+        get() = (testMongoClient as RemoteMongoClientImpl).dataSynchronizer
+    val testNetworkMonitor: BaseStitchIntTest.TestNetworkMonitor
+        get() = BaseStitchAndroidIntTest.testNetworkMonitor
+
+    fun getStitchAPIKey(): String {
+        return InstrumentationRegistry.getArguments().getString(stitchAPIKeyProp, "")
+    }
+
+    /**
+     * Get the uri for where mongodb is running locally.
+     */
+    fun getMongoDbUri(): String {
+        return InstrumentationRegistry.getArguments().getString(mongodbUriProp, "mongodb://localhost:26000")
+    }
+
+    override fun getStitchBaseURL(): String {
+        if (stitchTestHost.isNotEmpty()) {
+            return stitchTestHost
+        }
+        return InstrumentationRegistry.getArguments().getString(
+            "test.stitch.baseURL",
+            "http://10.0.2.2:9090"
+        )
+    }
+
+    override fun getAppClientConfigurationBuilder(): StitchAppClientConfiguration.Builder {
```

Comment:
Im worried that the context won't see this and this won't be used...

Reply:
Also, I'm pretty sure using stitchProd never used the transport (my fault)

Reply:
Why are you worried about that??

Reply:
Did you try it to see if it is working? I would actually imagine for local stitch it still works, but I need to add in using the new transport when connecting to prod because I dont think we currently do that.